[完了] タスクの一覧ページ作成

mikiya1130 opened the pull request:
- 変数にタスクを用意し、とりあえず表示させるところまで
- 文字列が長い場合は、溢れる分を省略記号に置き換え

close #7

mikiya1130 committed:
refactor: 余計な差分を防止するため

src ディレクトリ下のファイルにフォーマットを適用

mikiya1130 committed:
feat: タスク一覧を表示するため

mikiya1130 committed:
refactor: デフォルトのファイルが不必要になったため

mikiya1130 committed:
feat: 長いタスク名に対応するため

3点リーダーで省略して表示するように変更

watagit commented on todo-app/src/App.tsx:
```diff
@@ -1,26 +1,29 @@
+    <Container maxW="xl" centerContent>
+      {tasks.map((task: Task) => (
+        <Center p={3} m={1} w="100%" bg="blue.100">
+          <Text isTruncated>{task.text}</Text>
```
`isTruncated`とかあったんだ、、！知らなかった
便利だこれは 😆

watagit commented on todo-app/src/App.tsx:
```diff
@@ -1,26 +1,29 @@
-    </div>
+    <Container maxW="xl" centerContent>
+      {tasks.map((task: Task) => (
+        <Center p={3} m={1} w="100%" bg="blue.100">
```
![must](https://img.shields.io/badge/review-must-red)
`map`で配列を展開する際は、React が配列内の要素を識別できるようにするために`key`を指定しましょう 👍 

[[React チュートリアル] リストと key](https://ja.reactjs.org/docs/lists-and-keys.html#keys)

mikiya1130 replied:
https://github.com/bislab-temporary/react-study-morisaki/pull/14/commits/edde35000dc63e2f56f53335eb8e3352c31f6177
後回しにしていて、ここのコミットで追加しました。
どう対応するのがベストですか？(明日お願いします。)

watagit commented on todo-app/src/App.tsx:
```diff
@@ -1,26 +1,29 @@
+import { Center } from "@chakra-ui/layout";
+
+const App = () => {
+  interface Task {
+    text: string;
+  }
```
![imo](https://img.shields.io/badge/review-imo-yellow)
TypeScript で型を宣言する際は個人的には`type`がおすすめです！
`interface`の特徴として、**拡張性**が挙げられますが、React アプリの開発においては経験上メリットはないように思います。むしろ、容易に型を拡張できてしまうというデメリットになりかねないと思っています 🤔 
とはいいつつも、このあたりは個人の好みではあるので、こだわり等ある場合は`interface`で全く問題ないです 👍 

[[Zenn] interfaceとtypeの違い、そして何を使うべきかについて](https://zenn.dev/luvmini511/articles/6c6f69481c2d17#3.-%E3%81%A9%E3%81%A1%E3%82%89%E3%82%92%E4%BD%BF%E3%81%86%EF%BC%9F)

mikiya1130 replied:
調べて先に`interface`が出てきただけで、特にこだわりありません。
`type`で書き換えておきます。
参考リンクもありがとうございます。

watagit commented on todo-app/src/App.tsx:
```diff
@@ -1,26 +1,29 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { Container, Text } from "@chakra-ui/layout";
+import { Center } from "@chakra-ui/layout";
```
![nits](https://img.shields.io/badge/review-nits-lightgrey)
同じパッケージからのインポートなので、ひとつにまとめちゃいましょう！
```ts
import { Container, Text, Center } from "@chakra-ui/layout";
```

mikiya1130 committed:
refactor: 意図しない拡張を防ぐため

型の宣言を interface から type に変更
https://github.com/bislab-temporary/react-study-morisaki/pull/10#discussion_r719094454 への対応

mikiya1130 committed:
refactor: 記述を簡潔にするため

同じパッケージからのインポートを1つにまとめる
https://github.com/bislab-temporary/react-study-morisaki/pull/10#discussion_r719096085 への対応

mikiya1130 committed:
fix: コンフリクト防止のため、コードレビューへの対応を取り消すため

mikiya1130 committed:
Merge branch 'todo-app/feature/7_create_task_list_page' of https://github.com/bislab-temporary/react-study-morisaki into todo-app/feature/7_create_task_list_page
